var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -44,15 +45,35 @@ class Boost(Package):
     version('1.34.1', '2d938467e8a448a2c9763e0a9f8ca7e5')
     version('1.34.0', 'ed5b9291ffad776f8757a916e1726ad0')
 
+    libs = ['chrono', 
+        'date_time', 
+        'filesystem', 
+        'iostreams', 
+        'random', 
+        'regex', 
+        'serialization', 
+        'signals', 
+        'system', 
+        'thread', 
+        'wave', 
+        'mpi', 
+        'python']
```

Comment:
I think the spack way to do this would be:

`spack install boost +python ^python@3.1`

And it would automatically determine the config arg from the python version you asked for.